### PR TITLE
Add javascript templates support

### PR DESCRIPTION
**Pull request by baor**
Add a new field Response.JsTemplate which stores javascript code for generation response body.
It is possible to use plain text value as well as base64 string

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o gozzmock_bin .
 # Run stage
 FROM scratch
 
-MAINTAINER Travix
+LABEL maintainer="Travix"
```
Comment (Prutswonder) [inline Dockerfile]: All dockerfiles I've seen still have `MAINTAINER Travix`. Is this the preferred way of labeling?

File: handler.go [hunk 2]
```diff
@@ -105,7 +105,11 @@ func createResponseFromExpectation(w http.ResponseWriter, resp *ExpectationRespo
 		}
 	}
 	w.WriteHeader(resp.HTTPCode)
-	w.Write([]byte(resp.Body))
+	if len(resp.JsTemplate) > 0 {
+		w.Write([]byte(JsTemplateCreateResponseBody(resp.JsTemplate, req)))
+	} else {
+		w.Write([]byte(resp.Body))
```
Comment (Prutswonder) [inline handler.go]: Though there's nothing wrong with the syntax, code readability suggests that the preferred flow should take place as much as possible outside of conditions. What you'll usually see is that the not-so-ideal path is contained in an if-statement and contains a return. Then the ideal path doesn't need to be in the `else` part. 😉

File: jstemplate.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+// JsTemplateCreateResponseBody creates response body as string based on template and incoming request
+func JsTemplateCreateResponseBody(tmpl string, req *ExpectationRequest) string {
+	fLog := log.With().Str("function", "JsTemplateCreateResponseBody").Logger()
+
+	decodedTmpl, err := templateDecodeFromBase64(tmpl)
+	if err != nil {
+		fLog.Warn().Err(err)
+		decodedTmpl = tmpl
```
Comment (Prutswonder) [inline jstemplate.go]: This looks a bit risky. Is it because `tmpl ` can sometimes not be base64-encoded?

File: jstemplate_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		Body: `{
+			"a": [
+				{
+					"b": "bv1"
+				}
+				]}`}
+
+	tmpl := base64.StdEncoding.EncodeToString([]byte(`
+		var response = {"response": JSON.parse(request.Body)["a"][0]["b"]};
+		JSON.stringify(response)`))
```
Comment (Prutswonder) [inline jstemplate_test.go]: Missing semicolon after the stringify command.

File: Dockerfile (hunk 1 above)
Comment (baor) [reply Dockerfile]: Seems like it is an official way to replace MAINTAINER

https://docs.docker.com/engine/reference/builder/

>MAINTAINER (deprecated)

>To set a label corresponding to the MAINTAINER field you could use:

>LABEL maintainer="SvenDowideit@home.org.au"

File: jstemplate_test.go (hunk 4 above)
Comment (baor) [reply jstemplate_test.go]: this is my favorite 😀

File: jstemplate.go (hunk 3 above)
Comment (baor) [reply jstemplate.go]: Initially, I though about support both encoded and plain text templates.
Now I think plain text support doesn't worth it due to complexity of escaping.

File: handler.go [hunk 5]
```diff
@@ -94,18 +94,31 @@ func (context *Context) HandlerDefault(w http.ResponseWriter, r *http.Request) {
...
 		}
 	}
+
+	resposneBody := resp.Body
+	if len(resp.JsTemplate) > 0 {
+		var err error
+		resposneBody, err = JsTemplateCreateResponseBody(resp.JsTemplate, req)
+		if err != nil {
+			fLog.Error().Err(err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (baor) [inline handler.go]: I return error to the client, because client is responsible for JsTemplate content

**Pull request merged**